pkg/api/customization/pipeline/pipeline_execution.go

```diff
@@ -1,27 +1,39 @@
 package pipeline
 
 import (
-	"fmt"
-	"github.com/pkg/errors"
 	"github.com/rancher/norman/api/access"
 	"github.com/rancher/norman/httperror"
 	"github.com/rancher/norman/types"
-	"github.com/rancher/types/client/management/v3"
+	"github.com/rancher/rancher/pkg/clustermanager"
+	"github.com/rancher/rancher/pkg/pipeline/utils"
+	"github.com/rancher/rancher/pkg/ref"
+	"github.com/rancher/types/apis/project.cattle.io/v3"
+	"github.com/rancher/types/client/project/v3"
 	"net/http"
+	"time"
 )
 
 type ExecutionHandler struct {
+	ClusterManager *clustermanager.Manager
+
+	PipelineLister          v3.PipelineLister
+	PipelineExecutionLister v3.PipelineExecutionLister
+	PipelineExecutions      v3.PipelineExecutionInterface
 }
 
-func (h ExecutionHandler) ExecutionFormatter(apiContext *types.APIContext, resource *types.RawResource) {
-	resource.AddAction(apiContext, "rerun")
-	resource.AddAction(apiContext, "stop")
+func (h *ExecutionHandler) ExecutionFormatter(apiContext *types.APIContext, resource *types.RawResource) {
+	if e, ok := resource.Values["executionState"].(string); ok && e != utils.StateBuilding && e != utils.StateWaiting {
```

Comment:
do we need to check queue and pending status?

Reply:
Yes, will update.